test/parallel/test-module-customConditions.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+// Flags: -C additional
+
+require('../common');
+const assert = require('node:assert');
+const { customConditions } = require('module');
+
+assert.deepStrictEqual(customConditions, new Set(['additional']));
```

Comment:
Minor nit: we should also have a test that verifies that passing no conditions results in an empty set.

Reply:
I've added both tests.